modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2303,7 +2302,6 @@ public CodegenProperty fromProperty(String name, Schema p) {
             //    property.baseType = getSimpleRef(p.get$ref());
             //}
             // --END of revision
-            property.isModel = ModelUtils.isModel(p);
```

Comment:
If you're changing the template to not use `isModel`, we should keep this as the change affects all generators. See my resolution in #3855, which sets `isModel` as expected (after the naive attempt done in `setNonArrayMapProperty`.